### PR TITLE
TPP-00041: Register flow from onboarding and create account

### LINKED ISSUE #41: TPP-00041: Register flow from onboarding and create account
> As a user of Menstruation Nation, I want to be able to register, or register after onboarding if I wish to create an account

**Technical Information:**

- Add a red "Register" button on first onboarding screen above "Get Started" 
- clicking "Register" should lead to registration pages where we prompt user for their email address and then ask them to create a password. You don't need to do form cleaning for the input, so continue button should always be enabled 
- Ensure user can go back using left pointing arrow on top left "<" 
- Last page that says "Success" when you close it will lead to the month view calendar
- Helen has worked on the first few onboarding pages where she created  "How long does your period usually last", "When did your period last start", and "What symptoms do you want to track" (ticket here: https://github.com/uoftblueprint/the-period-purse/issues/20) 
- After those onboarding screen, we want it to go to "Would you like to create an account so your data is backed up?"
- Two buttons, "Register" and "continue as guest"
- Clicking "Continue as guest" leads to the "You're all set!" confirmation page. This page has an x at the top to close it. When it closes, should be on the mouth view calendar
- Clicking "Register" leads to the registration pages

**Styling Requirements:**

First onboarding screen: 
Screen Shot 2021-12-22 at 11 05 51 AM
- no image right now, can leave blank

Registration pages: 
Screen Shot 2021-12-22 at 1 31 13 PM
- no form cleaning, just ensure user can type in the input boxes
- continue button always enabled
- back button goes to previous page
- "Resend email" button does nothing 

Confirmation page: 
Screen Shot 2021-12-22 at 11 06 00 AM
- For the number of days, symptoms to log, just put placeholders there

### PR DESCRIPTION
**Pull request by helen-li**
# Description

* Modified the "Welcome" page with a new button for registration
* Modified the 3 existing onboarding pages to match the updated designs on Figma
* Added the last onboarding page ("Backup"), 2 registration pages for email and password, success page for registration, and the "You're all set!" confirmation page with placeholders
* Created a new components folder inside the onboarding folder to organize all the styled components in one place 
* Added new svg files for the icons I used

<p align="center">
  
  
  
  
  
  
  
  
  
  
</p>

Fixes #41 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tpp-app/.gitignore [hunk 1]
```diff
@@ -53,3 +53,5 @@ buck-out/
 # Expo
 .expo/
 web-build/
+
+/src/onboarding/LoadingScreen.js
```
Comment (kenneth-miura) [inline tpp-app/.gitignore:57]: why is this in the gitignore?

File: tpp-app/src/onboarding/Welcome.js [hunk 2]
```diff
@@ -1,42 +1,49 @@
...
-import { MyTabs as MainPage } from '../../App';
+import Backup from './Backup';
+import Registration from './Registration'
+import Password from './Password';
+import { MainPage } from '../../App';
+import Success from './Success';
+import Confirmation from './Confirmation';
+import { WideButton } from './components/ButtonComponents';
 
 export const STACK_SCREENS = {
```
Comment (kenneth-miura) [inline tpp-app/src/onboarding/Welcome.js:18]: what's the point of this if the keys & values are same? Is it so it's explicit (via the variable name) that these are screens? Not really opposed, just confused

**Review by kenneth-miura — COMMENTED** [review]
I like the way u created the stuff in the components folder. Think some of those are prob reusable elsewhere, but in that case we can just move em around.

File: tpp-app/.gitignore (hunk 1 above)
Comment (helen-li) [reply tpp-app/.gitignore:57]: LoadingScreen.js is the file for the splash / loading screen that I created when working on #20. At the time Chloe suggested that I don't link it up with anything yet and add it to .gitignore for now.

File: tpp-app/src/onboarding/Welcome.js (hunk 2 above)
Comment (helen-li) [reply tpp-app/src/onboarding/Welcome.js:18]: I took Roy's suggestion from my last ticket, this is mostly for avoiding spelling mistakes when navigating to the pages in the future.

**Comment by helen-li** [comment]
> I like the way u created the stuff in the components folder. Think some of those are prob reusable elsewhere, but in that case we can just move em around.

Yeppie we can def reuse some of the components elsewhere beyond these pages! I just wasn't too sure if we already had a structure in mind for how to organize styled things.

File: tpp-app/src/onboarding/Welcome.js (hunk 2 above)
Comment (kenneth-miura) [reply tpp-app/src/onboarding/Welcome.js:18]: yeah that makes sense

File: tpp-app/.gitignore (hunk 1 above)
Comment (kenneth-miura) [reply tpp-app/.gitignore:57]: cool & hip

**Review by kenneth-miura — APPROVED** [review]
looks good!

**Review by edwardhuahan — APPROVED** [review]
This looks great! I like how organized your code is

**Pull request merged**